typescript/src/promotional-offers/appleFetchOfferDetails.ts

```diff
@@ -0,0 +1,87 @@
+import 'source-map-support/register'
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { HTTPResponses } from '../models/apiGatewayHttp';
+import * as crypto from "crypto";
+import { getConfigValue } from "../utils/ssmConfig";
+
+interface HttpRequestPayload {
+    username: string,
+    productIdentifier: string,
+    offerIdentifier: string
+}
+
+interface Response {
+    nonce: string,
+    timestamp: number,
+    keyIdentifier: string,
+    signature: string
+}
+
+async function payloadToResponse(payload: HttpRequestPayload): Promise<Response> {
+
+    // https://developer.apple.com/documentation/storekit/in-app_purchase/original_api_for_in-app_purchase/subscriptions_and_offers/generating_a_signature_for_promotional_offers
+    // appBundleId         : (from ssm parameter store)
+    // keyIdentifier       : (from ssm parameter store)
+    // productIdentifier   : (from request payload)
+    // offerIdentifier     : (from request payload)
+    // applicationUsername : (from request payload as username)
+    // nonce               : (server generated)
+    // timestamp           : (server generated)
+
+    
+    // Notes:
+    //
+    //   1. The appBundleId is stored in parameter store because it's different in
+    //      CODE and PROD, but should be fairly stable.
+    //
+    //   2. The private key can be regenerated by one of the apps developers.
+    //
+    //   3. At the moment we are using the same key for CODE and PROD, but the code is written
+    //      to allow for them to be different.
+    //
+    //   4. If one day a key needs to be replaced, then both the key and its keyIdentifier will
+    //      have to be updated in AWS.
+
+    const appBundleId = await getConfigValue<string>("promotional-offers-appBundleId");
+    const keyIdentifier = await getConfigValue<string>("promotional-offers-keyIdentifier");
+    const productIdentifier = payload.productIdentifier;
+    const offerIdentifier = payload.offerIdentifier;
+    const applicationUsername = payload.username;
+    const nonce = crypto.randomUUID().toLowerCase();
+    const timestamp = Date. now(); // generates the currenty unixtime in milliseconds
```

Comment:
```suggestion
    const timestamp = Date.now(); // generates the current unixtime in milliseconds
```

you could probably leave out this comment

Reply:
Thanks. I definitively need glasses 😬